test/suite/test_assert06.py

```diff
@@ -0,0 +1,384 @@
+
+        # Create the table without the key consistency checking turned on.
+        # Create a few items breaking the rules. Then alter the setting and
+        # verify the inconsistent usage is detected.
+        self.session.create(uri, 'key_format=S,value_format=S')
+        # Insert a data item at timestamp 2.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value2'
+        self.apply_timestamps(2)
+        self.session.commit_transaction()
+        c.close()
+
+        # Modify the data item at timestamp 1.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value1'
+        self.apply_timestamps(1)
+        self.session.commit_transaction()
+        c.close()
+
+        # Insert a non-timestamped item. Then modify with a timestamp. And
+        # again modify without a timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots'
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value2'
+        self.apply_timestamps(2)
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots2'
+        self.session.commit_transaction()
+        c.close()
+
+        # We must move the oldest timestamp forward in order to alter.
+        # Otherwise alter closing the file will fail with EBUSY.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(2))
+
+        # Now alter the setting and make sure we detect incorrect usage.
+        self.session.alter(uri, cfg_on)
+
+        # Detect decreasing timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value5'
+        self.apply_timestamps(5)
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value4'
+        self.apply_timestamps(4)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(), msg_ooo)
+        c.close()
+
+        # Detect not using a timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value_nots3'
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(), msg_usage)
+        c.close()
+
+        # Detect using a timestamp on the non-timestamp key.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots3'
+        self.session.commit_transaction()
+        c.close()
```

Comment:
Do we need this operation here? At first I was confused thinking that you forgot to "use the timestamp" in the lines above. I eventually saw the lines below. If the operation above is not needed, remove it. If it is needed, please add a comment like: "Detect using a timestamp on the non-timestamp key. We first must perform a non-timestamped operation so that..."